.github/workflows/main.yaml

```diff
@@ -16,30 +16,27 @@ jobs:
         use-verbose-mode: 'yes'
         file-path: './README.md'
     
-  testing:
+  formatting:
     runs-on: ubuntu-22.04
-    name: Run test suites
+    name: Check formatting
     steps:
     - uses: actions/checkout@v3.1.0
     - name: Install nix
       uses: cachix/install-nix-action@v18
       with:
         extra_nix_config: |
           access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
-    - name: Processes backend test
-      run: nix develop .#lightweight --command cabal test smtlib-backends-process
-    - name: Z3 backend test
-      run: nix develop .#lightweight --command cabal test smtlib-backends-z3
+    - run: nix flake check
```

Comment:
Something like `nix flake check .#formatting` would be more explicit, and we can always set `checks.default = self.checks.formatting` if we want to stick with being able to just call `nix flake shell` in the CLI.